### PR TITLE
feat(worker): add worker deployment

### PR DESCRIPTION
**Pull request by Cliftonz**
### What change does this PR introduce?
Add worker deployment

### CODE REVIEW

**Review by LetItRock — APPROVED** [review]

File: .github/workflows/dev-deploy-worker.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  push:
+    branches:
+      - next
+      - main
+    paths:
+      - 'package.json'
+      - 'yarn.lock'
+      - 'apps/api/**'
+      - 'libs/dal/**'
+      - 'libs/shared/**'
```
Comment (scopsy) [inline .github/workflows/dev-deploy-worker.yml:16]: Let's add here also:
- application generic
- api

File: .github/workflows/dev-deploy-worker.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  test_worker:
+    strategy:
+      matrix:
+        name: ['novu/worker', 'novu/worker-ee']
+    uses: ./.github/workflows/reusable-worker-e2e.yml
+    with:
+      ee: ${{ contains (matrix.name,'ee') }}
+
+  deploy_dev_worker:
+    deploy_dev_webhook:
```
Comment (scopsy) [inline .github/workflows/dev-deploy-worker.yml:30]:
```suggestion
  deploy_dev_worker:
```

Indentation for the whole block needs to be fixed and also this duplicate should be removed

File: .github/workflows/dev-deploy-worker.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      if: "!contains(github.event.head_commit.message, 'ci skip')"
+      # The type of runner that the job will run on
+      runs-on: ubuntu-latest
+      needs: test_webhook
```
Comment (scopsy) [inline .github/workflows/dev-deploy-worker.yml:34]:
```suggestion
      needs: test_worker
```

File: .github/workflows/dev-deploy-worker.yml [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        deployments: write
+      steps:
+        - uses: actions/checkout@v3
+        - uses: ./.github/actions/setup-project
+
+        - name: Build, tag, and push image to Amazon ECR
+          id: build-image
+          env:
+            REGISTRY_OWNER: novuhq
+            DOCKER_NAME: novu/worker
```
Comment (scopsy) [inline .github/workflows/dev-deploy-worker.yml:49]:
```suggestion
            DOCKER_NAME: novu/worker-ee
```

For dev we only build EE

File: .github/workflows/dev-deploy-worker.yml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+      needs: test_webhook
+      timeout-minutes: 80
+      environment: Development
+      permissions:
+        contents: read
+        packages: write
+        deployments: write
+      steps:
+        - uses: actions/checkout@v3
+        - uses: ./.github/actions/setup-project
```
Comment (scopsy) [inline .github/workflows/dev-deploy-worker.yml:43]: I'm lacking here the prestage of copying the npm rc cloud or passing it in the build command as a param

File: .github/workflows/prod-deploy-worker.yml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+          docker run --network=host --name worker -dit --env NODE_ENV=test ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG
+          docker run --network=host appropriate/curl --retry 10 --retry-delay 5 --retry-connrefused http://localhost:3004/v1/health-check | grep 'ok'
+
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:prod
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:latest
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG
+          echo "::set-output name=IMAGE::ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG"
+
+  deploy_prod_worker_eu:
+    if: contains( ${{ matrix.name }} , 'ee')
```
Comment (scopsy) [inline .github/workflows/prod-deploy-worker.yml:77]: This is not a valid syntax from github actions and will fail when run. We have a similar problem in prod pipelines, there I hot fixed it by disabling the matrix above until we find a long term solution

File: .github/workflows/prod-deploy-worker.yml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+  deploy_prod_worker_us:
+    if: contains( ${{ matrix.name }} , 'ee')
+    needs:
+      - deploy_prod_worker_eu
+      - build_prod_image
+    uses: ./.github/workflows/reusable-app-service-deploy.yml
+    secrets: inherit
+    with:
+      service_name: api
```
Comment (scopsy) [inline .github/workflows/prod-deploy-worker.yml:94]:
```suggestion
      service_name: worker
```

File: .github/workflows/prod-deploy-worker.yml [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+      deployments: write
+      id-token: write
+    strategy:
+      matrix:
+        name: ['novu/worker', 'novu/worker-ee']
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-project
+
+      - name: build api
```
Comment (scopsy) [inline .github/workflows/prod-deploy-worker.yml:38]:
```suggestion
      - name: build worker
```

**Review by scopsy — CHANGES_REQUESTED** [review]

File: .github/workflows/dev-deploy-worker.yml (hunk 1 above)
Comment (LetItRock) [reply .github/workflows/dev-deploy-worker.yml:16]: added a few more from deps

File: .github/workflows/prod-deploy-worker.yml (hunk 6 above)
Comment (LetItRock) [reply .github/workflows/prod-deploy-worker.yml]: fixed

**Review by scopsy — APPROVED** [review]

**Pull request merged**